proofs/eo/cpc/Cpc.eo

```diff
@@ -391,6 +391,71 @@
   :conclusion (= a b)
 )
 
+;;;;; ProofRule::ANNIHILATE
+
+; program: $is_annihilate_rec
+; args:
+; - f (-> T U U): The type of function we are searching over, expected to be right-assoc-nil.
+; - t U: The term to compute whether it annilates.
+; - zero U: The zero element of f.
+; return: >
+;   true, if zero is found as a subterm of t beneath f-applications.
+(program $is_annihilate_rec ((T Type) (U Type) (f (-> T U U)) (a T) (b U) (zero U))
+  ((-> T U U) U U) Bool
+  (
+    (($is_annihilate_rec f zero zero)     true)
+    (($is_annihilate_rec f (f a b) zero)  (eo::ite ($is_annihilate_rec f a zero)
+                                            true
+                                            ($is_annihilate_rec f b zero)))
```

Comment:
I think the if-then-else expression can be simplified to a disjunction (unless short-circuiting is not supported)

Reply:
`eo::ite` is the only operator that "short-circuits", which is why this is written this way.